plugins/zenoh-plugin-trait/src/compatibility.rs

```diff
@@ -0,0 +1,258 @@
+    /// If comparison failed, the corresponding field in both structs is kept as is.
+    /// This allows not only to check compatibility, but also point to exact reasons of incompatibility.
+    pub fn compare(&mut self, other: &mut Self) -> bool {
+        let mut result = true;
+        Self::compare_field_fn(
+            &mut result,
+            &mut self.rust_version,
+            &mut other.rust_version,
+            RustVersion::are_compatible,
+        );
+        Self::compare_field(
+            &mut result,
+            &mut self.vtable_version,
+            &mut other.vtable_version,
+        );
+        Self::compare_field(
+            &mut result,
+            &mut self.start_args_version,
+            &mut other.start_args_version,
+        );
+        Self::compare_field(
+            &mut result,
+            &mut self.instance_version,
+            &mut other.instance_version,
+        );
+        // TODO: here we can later implement check for plugin version range compatibility
+        Self::compare_field(
+            &mut result,
+            &mut self.plugin_version,
+            &mut other.plugin_version,
+        );
+        Self::compare_field(
+            &mut result,
+            &mut self.plugin_long_version,
+            &mut other.plugin_long_version,
+        );
+        result
+    }
+
+    // Utility function for compare single field
+    fn compare_field_fn<T, F: Fn(&T, &T) -> bool>(
+        result: &mut bool,
+        a: &mut Option<T>,
+        b: &mut Option<T>,
+        compare: F,
+    ) {
+        let compatible = if let (Some(a), Some(b)) = (&a, &b) {
+            compare(a, b)
+        } else {
+            true
+        };
+        if compatible {
+            *a = None;
+            *b = None;
+        } else {
+            *result = false;
+        }
+    }
+    fn compare_field<T: PartialEq>(result: &mut bool, a: &mut Option<T>, b: &mut Option<T>) {
+        Self::compare_field_fn(result, a, b, |a, b| a == b);
+    }
+}
+
+impl Display for Compatibility {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Some(rust_version) = &self.rust_version {
+            writeln!(f, "Rust version:\n{}", rust_version)?;
+        }
+        if let Some(vtable_version) = &self.vtable_version {
+            writeln!(f, "VTable version:\n{}", vtable_version)?;
+        }
+        if let Some(start_args_version) = &self.start_args_version {
+            writeln!(f, "StartArgs version:\n{}", start_args_version)?;
+        }
+        if let Some(instance_version) = &self.instance_version {
+            writeln!(f, "Instance version:\n{}", instance_version)?;
+        }
+        if let Some(plugin_version) = &self.plugin_version {
+            writeln!(f, "Plugin version: {}", plugin_version)?;
+        }
+        Ok(())
+    }
+}
+
+#[repr(C)]
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct RustVersion {
+    major: u64,
```

Comment:
Not a blocker for this PR but `u64` seems a lot for the version.

Reply:
fixed, seems like u32 for each field should be enough